workspaces/optic-engine-wasm/src/lib.rs

```diff
@@ -380,6 +380,28 @@ pub fn spec_resolve_response(
     .map_err(|err| JsValue::from(format!("responses could not be serialized: {:?}", err)))
 }
 
+#[wasm_bindgen]
+pub fn spec_endpoint_delete_commands(
+  spec: &WasmSpecProjection,
+  path_id: String,
+  method: String,
+) -> Result<String, JsValue> {
+  let endpoint_queries = spec.endpoint_queries();
+
+  let commands = endpoint_queries
+    .delete_endpoint_commands(&path_id, &method)
+    .ok_or_else(|| {
+      JsValue::from("delete endpoint commands could not be generated for unexisting endpoint")
+    })?;
+
+  serde_json::to_string(&commands).map_err(|err| {
+    JsValue::from(format!(
+      "delete enpoints commands could not be serialized: {:?}",
+      err
+    ))
+  })
```

Comment:
For simplicity's sake, we serialise the results to a string and require the consuming code to deserialise again. Note that this doesn't mean necessarily Spectacle: it could choose to pipe the results straight through to the UI, where serialisation has to happen anyway.  There's alternatives for creating Javascript objects, or returning constructs  in the shared WASM memory (which are basically opaque JS objects containing only a pointer into memory) , however:

- creating Javascript objects [isn't more performant or efficient](https://rustwasm.github.io/wasm-bindgen/reference/arbitrary-data-with-serde.html#an-alternative-approach-serde-wasm-bindgen)
- working with opaque data structures is a pain, especially once you try to debug or inspect them. After going that route with the ScalaJS stuff, we reckoned it's more straightforward for the JS side to always work with JSON / Javascript objects.